src/coreclr/jit/indirectcalltransformer.cpp

```diff
@@ -862,7 +862,7 @@ class IndirectCallTransformer
 
             JITDUMP("Direct call [%06u] in block " FMT_BB "\n", compiler->dspTreeID(call), block->bbNum);
 
-            CORINFO_METHOD_HANDLE  methodHnd = call->gtCallMethHnd;
+            CORINFO_METHOD_HANDLE  methodHnd = inlineInfo->guardedMethodHandle;
```

Comment:
I think this isn't strictly necessary...? 

Consider leaving as is and then asserting that the method that `impDevirtualizeCall` produces is the one we expected?

Reply:
I'll check with recent David's VM changes, but last time I checked this was an important part to fix asserts

Reply:
Just checked - we still assert in both VMs for that. The problem that we pass the base `MethodDesc` + more precise generic context that belongs to the `inlineInfo->guardedMethodHandle`